pkg/provisioners/channel_util.go

```diff
@@ -169,6 +187,40 @@ func CreateVirtualService(ctx context.Context, client runtimeClient.Client, chan
 	return virtualService, nil
 }
 
+// checkExpectedLabels checks the presence of expected labels and its values and return true
+// if all labels are found.
+func checkExpectedLabels(actual, expected map[string]string) bool {
+	for ke, ve := range expected {
+		if va, ok := actual[ke]; ok {
+			if strings.Compare(ve, va) == 0 {
+				continue
+			}
+		}
+		return false
+	}
+	return true
+}
+
+// addExpectedLabels adds expected labels
+func addExpectedLabels(actual, expected map[string]string) map[string]string {
+	consolidated := make(map[string]string, 0)
+	// First store all exisiting labels
+	for k, v := range actual {
+		consolidated[k] = v
+	}
+	// Second add all missing expected labels
+	for k, v := range expected {
+		if va, ok := consolidated[k]; ok {
```

Comment:
Shouldn't this always overwrite? If the value of the label is different than expected, don't we want to overwrite that value to be what we expect?

Reply:
yep, you are right, thanks for catching it.